yt_dlp/extractor/yandexvideo.py

```diff
@@ -268,35 +270,74 @@ class ZenYandexChannelIE(InfoExtractor):
+            'title': 'ТОК',
         },
         'playlist_mincount': 169,
     }, {
         'url': 'https://zen.yandex.ru/id/606fd806cc13cb3c58c05cf5',
         'info_dict': {
             'id': '606fd806cc13cb3c58c05cf5',
+            'description': 'md5:517b7c97d8ca92e940f5af65448fd928',
+            'title': 'AcademeG DailyStream',
         },
         'playlist_mincount': 657,
     }]
 
-    def _entries(self, id, url):
-        webpage = self._download_webpage(url, id)
-        data_json = self._parse_json(re.findall(r'var\s?data\s?=\s?({.+?})\s?;', webpage)[-1], id)
-        for key in data_json.keys():
-            if key.startswith('__serverState__'):
-                data_json = data_json[key]
-        items = list(try_get(data_json, lambda x: x['feed']['items'], dict).values())
-        more = try_get(data_json, lambda x: x['links']['more']) or None
+    def _entries(self, channel_title, server_state_json, server_settings_json):
+        items = list((try_get(server_state_json, lambda x: x['feed']['items'], dict) or {}).values())
+
+        if(len(items) == 0):
+            items = list(try_get(server_settings_json, lambda x: x['exportData']['items']))
+
+        prev_next_page_id = None
+        more = try_get(
+            server_state_json, lambda x: x['links']['more']
+        ) or try_get(
+            server_settings_json, lambda x: x['exportData']['more']['link']) or None
+
+        if more:
+            next_page_id = try_get(parse_qs(more), lambda x: x['next_page_id'][0]) or None
+        else:
+            next_page_id = None
+
         for page in itertools.count(1):
-            for item in items:
+            for item in filter(lambda x: x.get('type') == 'gif', items):
                 video_id = item.get('publication_id') or item.get('publicationId')
                 video_url = item.get('link')
                 yield self.url_result(video_url, ie=ZenYandexIE.ie_key(), video_id=video_id.split(':')[-1])
-            if not more:
+
+            if any([not more, len(items) == 0, next_page_id and next_page_id == prev_next_page_id, not next_page_id]):
                 break
-            data_json = self._download_json(more, id, note='Downloading Page %d' % page)
+
+            data_json = self._download_json(more, channel_title, note='Downloading Page %d' % page)
             items = data_json.get('items', [])
             more = try_get(data_json, lambda x: x['more']['link']) or None
 
+            prev_next_page_id = next_page_id
+            if more:
+                next_page_id = try_get(parse_qs(more), lambda x: x['next_page_id'][0]) or None
+            else:
+                next_page_id = None
+
     def _real_extract(self, url):
         id = self._match_id(url)
-        return self.playlist_result(self._entries(id, url), playlist_id=id)
+        webpage = self._download_webpage(url, id)
+        data_json = self._parse_json(re.findall(r'var\s?data\s?=\s?({.+?})\s?;', webpage)[-1], id)
```

Comment:
```suggestion
        data_json = self._parse_json(re.findall(r'var\s+data\s*=\s*({.+?})\s*;', webpage)[-1], id)
```
Why are you using findall and then taking just last? The order could easily change in future. Isn't there a better regex that can match just the item we need?